src/sql/orderbook/batch_rewards.sql

```diff
@@ -0,0 +1,44 @@
+WITH hashed_observations AS (
+SELECT
+    -- settlement
+    tx_hash,
+    -- settlement_observations
+    block_number,
+    effective_gas_price * gas_used AS execution_cost,
+    surplus,
+    fee,
+    -- auction_transaction
+    auction_id
+FROM settlement_observations so
```

Comment:
I guess this is a question for the PR that introduced the `settlement_observations` table, but just wanted to clarify here. Does this also contain the settlements that reverted?

Reply:
Not to my knowledge, how could it? There would be nothing to be observed, but this is a great point. we can get that information from dune, but I think there is a tendency to try and be able to compute this entirely from our own db. In that case one would have to monitor all call to the contract (I.e failed transactions) and this would have to be part of the backend services which it DEFINITELY is not currently.

Reply:
>  but I think there is a tendency to try and be able to compute this entirely from our own db. In that case one would have to monitor all call to the contract (I.e failed transactions) and this would have to be part of the backend services which it DEFINITELY is not currently.

Yes, I would expect that we do everything from our own db. And to me, although I might be completely wrong here, it makes sense that for every auction id that has a winning solution, we store all onchain details around it, if any. This would imply storing a tx hash (and whatever relevant information we need) even in the case of a revert, as this connects our offchain auction with the corresponding onchain activity, if any.

Reply:
For reverted transactions we would populate only `block_number, effective_gas_price, gas used` right?
Is this MUST HAVE for CIP20? Backend relies on emitted events so currently there is no way to index reverted transactions, implementing this would probably delay CIP20 for a few days. I would rather do it as a follow up.

Reply:
As discussed earlier today, it would suffice to add a column `block_deadline` to the settlement_scores table for which we can infer these non-existent values and still be able to assign a block number to the batch.